test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/Services/NuGetProjectManagerServiceTests.cs

```diff
@@ -533,6 +535,97 @@ public async Task GetUpdateActionsAsync_WithPackageReferenceProject_WhenUpdating
             }
         }
 
+        [Fact]
+        private async Task GetTransitivePackageOriginAsync_WithPackageReferenceProject_OneTransitiveReferenceAsync()
+        {
+            string projectName = Guid.NewGuid().ToString();
+            string projectId = projectName;
+            var projectSystemCache = new ProjectSystemCache();
+            IVsProjectAdapter projectAdapter = Mock.Of<IVsProjectAdapter>();
+
+            using (TestDirectory testDirectory = TestDirectory.Create())
+            {
+                Initialize();
+
+                // Prepare: Create project
+                string projectFullPath = Path.Combine(testDirectory.Path, $"{projectName}.csproj");
+
+                var prProject = CpsPackageReferenceProjectTests.CreateCpsPackageReferenceProject(projectName, projectFullPath,
+                    projectSystemCache);
+
+                ProjectNames projectNames = CpsPackageReferenceProjectTests.GetTestProjectNames(projectFullPath, projectName);
```

Comment:
It looks this one is for `CpsPackageReferenceProject`.
Do we need `LegacyPackageReference` type unit test too?

Reply:
Yes. Added test.